localstack/utils/aws/message_forwarding.py

```diff
@@ -29,63 +27,23 @@
 AUTH_OAUTH = "OAUTH_CLIENT_CREDENTIALS"
 
 
-def lambda_result_to_destination(
-    func_details: LambdaFunction,
-    event: Dict,
-    result: InvocationResult,
-    is_async: bool,
-    error: Optional[InvocationException],
-):
-    if not func_details.destination_enabled():
-        return
-
-    payload = {
-        "version": "1.0",
-        "timestamp": timestamp_millis(),
-        "requestContext": {
-            "requestId": long_uid(),
-            "functionArn": func_details.arn(),
-            "condition": "RetriesExhausted",
-            "approximateInvokeCount": 1,
-        },
-        "requestPayload": event,
-        "responseContext": {"statusCode": 200, "executedVersion": "$LATEST"},
-        "responsePayload": {},
-    }
-
-    if result and result.result:
-        try:
-            payload["requestContext"]["condition"] = "Success"
-            payload["responsePayload"] = json.loads(result.result)
-        except Exception:
-            payload["responsePayload"] = result.result
-
-    if error:
-        payload["responseContext"]["functionError"] = "Unhandled"
-        # add the result in the response payload
-        if error.result is not None:
-            payload["responsePayload"] = json.loads(error.result)
-        send_event_to_target(func_details.on_failed_invocation, payload)
-        return
-
-    if func_details.on_successful_invocation is not None:
-        send_event_to_target(func_details.on_successful_invocation, payload)
-
-
 def send_event_to_target(
     target_arn: str,
     event: Dict,
     target_attributes: Dict = None,
     asynchronous: bool = True,
-    target: Dict = {},
+    target: Dict = None,
```

Comment:
:+1: we should probably do some kind of analysis over the whole repo at some point to detect all mutable default parameters

Reply:
In this case it was used read only, but I decided to change it anyway, in case someone starts modifying it. I agree, but I think someone already did it once if I am not mistaken